### ISSUE TITLE
Report: chart is cut from the top

### ISSUE THREAD

**Issue by v-stepanov** [zalando-zmon/zmon-slo-metrics#59]
Right now the max-y value of chart is calculated according to SLO threshold not the maximum value of the chart. As a result - not the whole chart is visible - it is cut from the top. The points are are much higher than SLO threshold are not visible.

Example of problem: https://pages.github.bus.zalan.do/continuous-delivery/automata-service-level-reports/messaging-bus/nakadi/20170519-20170522/index.html

Is it expected behavior or a bug?
If it's a bug - I can contribute with a fix.

**Comment by mohabusama** [comment]
@v-stepanov Thanks for the report.

Right now we are refactoring a bit, but the report generation refactoring will come in a later stage.

PRs are welcome, if you contribute the fix we will merge.

**Comment by v-stepanov** [comment]
I tried to fix this problem but I faced another problem when doing that :)

The problem is that gnuplot uses not all points to build a chart but but some fixed amount [(300)](https://github.com/zalando-zmon/service-level-reporting/blob/master/plot.py#L41) if I switch to using all points - then the chart looks ugly :)

So the problem I have because of that is that when calculating max-y I check all data-points but the chart displays only 300 points and very often the maximum value I find is not displayed on chart so the chart-line itself is somewhere at the bottom and there is a lot of free space in the middle and the top. And this looks really bad:
![chart-587](https://cloud.githubusercontent.com/assets/13309177/26494260/30d0d832-421e-11e7-80f0-d0bc29b012a6.png)

I also tried to remove setting of min-y and max-y values so that gnuplot decides itself which range to use. But the result was the same again like on the picture - it seems that gnuplot also considers points that are not rendered. 

I will close this issue as I don't really see a good way to fix that. 
I'm fine with the way the chart is currently rendered.
